# V3 empirical study — 7-day hourly calibration

**This notebook is the computational source of truth.** The PDF in this folder is written from the same `payload` dict.

| Item | Setting |
|------|---------|
| Models | GBM, GARCH, Heston, Merton, GARCH–Merton, Heston–Merton |
| Companies | SPY, AAPL, MSFT |
| Windows | 12 third-Friday monthly expiries, 21 Oct 2022 – 15 Sep 2023 |
| Evaluation | **five NYSE sessions ending the Friday before expiry (hourly 09:59–15:59)** |
| Calibration | **7-day** lookback, **hourly** rolling, 1-minute RTH data |
| LSM horizon | remaining time **to window Friday 15:59**, not listed expiry |
| LSM grid | Δt = **5 minute(s)**, n_steps = remaining RTH / Δt |
| Paths | `10000`, seed `42` |
| Sample | one nearest-ATM listed call per hourly timestamp, DTE 7–60 |

Green / Mark 1 = lowest RMSE%. Friday 15:59 is scheduled but LSM is skipped (`remaining_horizon = 0`).


## Methodology

- Shared sample: nearest-ATM, DTE 7–60, listed expiry, as-of each hourly *t*. All six models price the same contracts.
- No look-ahead: parameters at *t* use only equity returns and listed quotes with timestamp ≤ *t*.
- V3 Heston: option-implied NLS; Euler uses *v_t* then updates variance.
- LSM on the full path cloud. Start every path at the observed *S_t*.


In [1]:
import sys
from pathlib import Path
from IPython.display import display, HTML
ROOT = Path.cwd()
for cand in [ROOT, *ROOT.parents]:
    scripts = cand / 'V3-Models_result' / 'scripts'
    if scripts.exists():
        sys.path.insert(0, str(scripts))
        break
import run_v3_7d_hourly_empirical_study as study
RECOMPUTE = False
payload = study.run_or_load(recompute=RECOMPUTE)
print('cells', len(payload['cells']), 'failures', payload['meta']['failures'])

## Shared sample *n*


In [1]:
display(HTML('shared n'))

Window,Eval Friday,SPY n,AAPL n,MSFT n
Oct 2022 2022-10-21,2022-10-14,34,34,34
Nov 2022 2022-11-18,2022-11-11,34,34,34
Dec 2022 2022-12-16,2022-12-09,34,34,34
Jan 2023 2023-01-20,2023-01-13,34,34,34
Feb 2023 2023-02-17,2023-02-10,34,34,34
Mar 2023 2023-03-17,2023-03-10,34,34,34
Apr 2023 2023-04-21,2023-04-14,34,34,34
May 2023 2023-05-19,2023-05-12,34,34,34
Jun 2023 2023-06-16,2023-06-09,34,34,34
Jul 2023 2023-07-21,2023-07-14,34,34,34


## SPY


### Table SPY-1  ·  Oct 2022  2022-10-21  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2022-10-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,73.94,9.1695,-9.1695,-69.89,57.7%,6
GARCH,73.18,9.0844,-9.0844,-69.23,56.2%,4
Heston,51.57,6.4959,-6.4959,-49.07,53.4%,2
Merton,73.90,9.1627,-9.1627,-69.84,55.5%,5
GARCH–Merton,72.86,9.0470,-9.0470,-68.94,52.5%,3
Heston–Merton,51.30,6.4592,-6.4592,-48.79,52.8%,1


### Table SPY-2  ·  Nov 2022  2022-11-18  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2022-11-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,99.78,10.5539,-10.5539,-99.78,1.0%,5
GARCH,99.78,10.5546,-10.5546,-99.78,0.9%,6
Heston,93.64,9.8641,-9.8641,-93.42,7.7%,2
Merton,99.77,10.5529,-10.5529,-99.77,1.0%,4
GARCH–Merton,99.68,10.5430,-10.5430,-99.68,1.3%,3
Heston–Merton,93.57,9.8560,-9.8560,-93.34,7.5%,1


### Table SPY-3  ·  Dec 2022  2022-12-16  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2022-12-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,9.1615,-9.1615,-100.00,0.0%,5
GARCH,100.00,9.1615,-9.1615,-100.00,0.0%,4
Heston,98.63,9.0347,-9.0347,-98.62,2.1%,1
Merton,100.00,9.1615,-9.1615,-100.00,0.0%,6
GARCH–Merton,100.00,9.1614,-9.1614,-100.00,0.0%,3
Heston–Merton,98.65,9.0365,-9.0365,-98.64,2.0%,2


### Table SPY-4  ·  Jan 2023  2023-01-20  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-01-20'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,9.1865,-9.1865,-100.00,0.0%,3
GARCH,100.00,9.1865,-9.1865,-100.00,0.0%,4
Heston,99.85,9.1714,-9.1714,-99.85,0.3%,1
Merton,100.00,9.1865,-9.1865,-100.00,0.0%,5
GARCH–Merton,100.00,9.1865,-9.1865,-100.00,0.0%,6
Heston–Merton,99.85,9.1717,-9.1717,-99.85,0.3%,2


### Table SPY-5  ·  Feb 2023  2023-02-17  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-02-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,9.1806,-9.1806,-100.00,0.0%,5
GARCH,100.00,9.1806,-9.1806,-100.00,0.0%,3
Heston,99.48,9.1323,-9.1323,-99.48,1.0%,1
Merton,100.00,9.1806,-9.1806,-100.00,0.0%,6
GARCH–Merton,100.00,9.1806,-9.1806,-100.00,0.0%,4
Heston–Merton,99.49,9.1332,-9.1332,-99.49,1.0%,2


### Table SPY-6  ·  Mar 2023  2023-03-17  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-03-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,95.73,8.1177,-8.1177,-95.33,12.1%,5
GARCH,95.15,8.0669,-8.0669,-94.74,13.8%,4
Heston,80.02,6.6883,-6.6883,-78.57,22.9%,2
Merton,95.76,8.1197,-8.1197,-95.36,12.2%,6
GARCH–Merton,94.98,8.0515,-8.0515,-94.56,13.7%,3
Heston–Merton,79.82,6.6742,-6.6742,-78.40,23.1%,1


### Table SPY-7  ·  Apr 2023  2023-04-21  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-04-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,8.7247,-8.7247,-100.00,0.0%,5
GARCH,100.00,8.7246,-8.7246,-100.00,0.0%,4
Heston,97.69,8.5107,-8.5107,-97.66,4.8%,2
Merton,100.00,8.7247,-8.7247,-100.00,0.0%,6
GARCH–Merton,100.00,8.7246,-8.7246,-100.00,0.0%,3
Heston–Merton,97.66,8.5081,-8.5081,-97.63,4.2%,1


### Table SPY-8  ·  May 2023  2023-05-19  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-05-19'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,7.7763,-7.7763,-100.00,0.0%,5
GARCH,100.00,7.7762,-7.7762,-100.00,0.0%,4
Heston,98.88,7.6820,-7.6820,-98.87,2.8%,2
Merton,100.00,7.7763,-7.7763,-100.00,0.0%,6
GARCH–Merton,100.00,7.7761,-7.7761,-100.00,0.0%,3
Heston–Merton,98.81,7.6760,-7.6760,-98.80,2.5%,1


### Table SPY-9  ·  Jun 2023  2023-06-16  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-06-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,5.6153,-5.6153,-100.00,0.0%,3
GARCH,100.00,5.6153,-5.6153,-100.00,0.0%,4
Heston,99.93,5.6108,-5.6108,-99.93,0.2%,1
Merton,100.00,5.6153,-5.6153,-100.00,0.0%,5
GARCH–Merton,100.00,5.6153,-5.6153,-100.00,0.0%,6
Heston–Merton,99.94,5.6115,-5.6115,-99.94,0.1%,2


### Table SPY-10  ·  Jul 2023  2023-07-21  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-07-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,7.0074,-7.0074,-100.00,0.0%,3
GARCH,100.00,7.0074,-7.0074,-100.00,0.0%,4
Heston,98.93,6.9253,-6.9253,-98.92,2.0%,2
Merton,100.00,7.0074,-7.0074,-100.00,0.0%,5
GARCH–Merton,100.00,7.0074,-7.0074,-100.00,0.0%,6
Heston–Merton,98.93,6.9254,-6.9254,-98.92,1.9%,1


### Table SPY-11  ·  Aug 2023  2023-08-18  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-08-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,7.9176,-7.9176,-100.00,0.0%,3
GARCH,100.00,7.9176,-7.9176,-100.00,0.0%,4
Heston,99.99,7.9171,-7.9171,-99.99,0.0%,1
Merton,100.00,7.9176,-7.9176,-100.00,0.0%,5
GARCH–Merton,100.00,7.9176,-7.9176,-100.00,0.0%,6
Heston–Merton,99.99,7.9171,-7.9171,-99.99,0.0%,2


### Table SPY-12  ·  Sep 2023  2023-09-15  ·  n = 34  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-09-15'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,6.1782,-6.1782,-100.00,0.0%,1
GARCH,100.00,6.1782,-6.1782,-100.00,0.0%,2
Heston,100.00,6.1782,-6.1782,-100.00,0.0%,3
Merton,100.00,6.1782,-6.1782,-100.00,0.0%,4
GARCH–Merton,100.00,6.1782,-6.1782,-100.00,0.0%,5
Heston–Merton,100.00,6.1782,-6.1782,-100.00,0.0%,6


## AAPL


### Table AAPL-1  ·  Oct 2022  2022-10-21  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2022-10-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,60.96,4.2521,-3.5283,-48.63,83.4%,4
GARCH,61.05,4.2558,-3.5316,-48.68,83.9%,6
Heston,45.43,3.0779,-2.3283,-32.34,58.8%,2
Merton,60.96,4.2524,-3.5267,-48.61,79.5%,5
GARCH–Merton,60.79,4.2383,-3.5141,-48.44,79.2%,3
Heston–Merton,45.28,3.0659,-2.3126,-32.13,56.7%,1


### Table AAPL-2  ·  Nov 2022  2022-11-18  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2022-11-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,95.97,5.6617,-5.6617,-95.62,10.8%,6
GARCH,94.69,5.5833,-5.5833,-94.33,18.1%,4
Heston,81.67,4.7360,-4.7360,-80.47,19.5%,2
Merton,95.96,5.6611,-5.6611,-95.61,10.8%,5
GARCH–Merton,94.26,5.5549,-5.5549,-93.88,18.5%,3
Heston–Merton,81.63,4.7327,-4.7327,-80.41,18.7%,1


### Table AAPL-3  ·  Dec 2022  2022-12-16  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2022-12-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,91.49,4.9914,-4.9914,-90.96,27.8%,5
GARCH,91.79,5.0086,-5.0086,-91.28,27.6%,6
Heston,68.25,3.6758,-3.6758,-67.08,32.4%,2
Merton,91.48,4.9911,-4.9911,-90.95,26.4%,4
GARCH–Merton,91.22,4.9776,-4.9776,-90.71,25.4%,3
Heston–Merton,68.07,3.6684,-3.6684,-66.95,33.0%,1


### Table AAPL-4  ·  Jan 2023  2023-01-20  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-01-20'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,99.58,5.8454,-5.8454,-99.57,1.5%,4
GARCH,99.60,5.8464,-5.8464,-99.59,1.6%,6
Heston,93.18,5.4510,-5.4510,-92.88,9.9%,1
Merton,99.58,5.8457,-5.8457,-99.58,1.4%,5
GARCH–Merton,99.52,5.8420,-5.8420,-99.52,1.5%,3
Heston–Merton,93.19,5.4524,-5.4524,-92.90,9.8%,2


### Table AAPL-5  ·  Feb 2023  2023-02-17  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-02-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,97.24,4.9648,-4.9648,-97.08,6.3%,4
GARCH,97.76,4.9909,-4.9909,-97.61,5.6%,6
Heston,83.29,4.2225,-4.2225,-82.27,16.4%,2
Merton,97.19,4.9620,-4.9620,-97.02,6.2%,3
GARCH–Merton,97.51,4.9776,-4.9776,-97.34,5.5%,5
Heston–Merton,83.27,4.2221,-4.2221,-82.26,16.8%,1


### Table AAPL-6  ·  Mar 2023  2023-03-17  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-03-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,77.49,3.1398,-3.1398,-75.24,45.6%,5
GARCH,77.63,3.1466,-3.1466,-75.42,46.6%,6
Heston,53.35,2.0299,-2.0260,-47.65,39.7%,2
Merton,77.39,3.1348,-3.1348,-75.11,43.5%,4
GARCH–Merton,77.03,3.1208,-3.1208,-74.77,42.1%,3
Heston–Merton,53.11,2.0207,-2.0172,-47.44,39.8%,1


### Table AAPL-7  ·  Apr 2023  2023-04-21  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-04-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,84.59,4.8063,-4.8063,-82.48,40.7%,6
GARCH,84.55,4.8040,-4.8040,-82.44,38.6%,4
Heston,71.20,4.0260,-4.0260,-69.16,42.6%,2
Merton,84.57,4.8044,-4.8044,-82.45,36.9%,5
GARCH–Merton,84.27,4.7892,-4.7892,-82.18,41.5%,3
Heston–Merton,71.18,4.0236,-4.0236,-69.12,41.3%,1


### Table AAPL-8  ·  May 2023  2023-05-19  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-05-19'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,99.29,3.7442,-3.7442,-99.28,2.1%,4
GARCH,99.41,3.7502,-3.7502,-99.41,1.8%,6
Heston,95.05,3.5477,-3.5477,-94.73,6.6%,1
Merton,99.27,3.7428,-3.7428,-99.25,2.0%,3
GARCH–Merton,99.31,3.7449,-3.7449,-99.29,2.4%,5
Heston–Merton,95.07,3.5488,-3.5488,-94.76,6.7%,2


### Table AAPL-9  ·  Jun 2023  2023-06-16  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-06-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,4.6852,-4.6852,-100.00,0.0%,5
GARCH,100.00,4.6851,-4.6851,-100.00,0.0%,4
Heston,98.06,4.5584,-4.5584,-97.99,2.6%,1
Merton,100.00,4.6853,-4.6853,-100.00,0.0%,6
GARCH–Merton,100.00,4.6850,-4.6850,-100.00,0.0%,3
Heston–Merton,98.06,4.5589,-4.5589,-97.99,2.7%,2


### Table AAPL-10  ·  Jul 2023  2023-07-21  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-07-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,4.5360,-4.5360,-100.00,0.0%,3
GARCH,100.00,4.5360,-4.5360,-100.00,0.0%,4
Heston,99.64,4.5210,-4.5210,-99.64,0.7%,1
Merton,100.00,4.5360,-4.5360,-100.00,0.0%,5
GARCH–Merton,100.00,4.5360,-4.5360,-100.00,0.0%,6
Heston–Merton,99.64,4.5212,-4.5212,-99.64,0.7%,2


### Table AAPL-11  ·  Aug 2023  2023-08-18  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-08-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,3.9147,-3.9147,-100.00,0.0%,4
GARCH,100.00,3.9147,-3.9147,-100.00,0.0%,5
Heston,99.65,3.8999,-3.8999,-99.65,0.6%,2
Merton,100.00,3.9147,-3.9147,-100.00,0.0%,6
GARCH–Merton,100.00,3.9147,-3.9147,-100.00,0.0%,3
Heston–Merton,99.65,3.8999,-3.8999,-99.64,0.6%,1


### Table AAPL-12  ·  Sep 2023  2023-09-15  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-09-15'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,4.0309,-4.0309,-100.00,0.0%,3
GARCH,100.00,4.0309,-4.0309,-100.00,0.0%,4
Heston,99.94,4.0289,-4.0289,-99.94,0.1%,1
Merton,100.00,4.0309,-4.0309,-100.00,0.0%,5
GARCH–Merton,100.00,4.0309,-4.0309,-100.00,0.0%,6
Heston–Merton,99.96,4.0293,-4.0293,-99.96,0.1%,2


## MSFT


### Table MSFT-1  ·  Oct 2022  2022-10-21  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2022-10-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,70.28,7.3746,-4.2334,-37.51,70.0%,6
GARCH,70.21,7.3670,-4.2277,-37.46,69.4%,4
Heston,55.07,5.6718,-2.4708,-22.17,62.0%,2
Merton,70.27,7.3709,-4.2324,-37.50,70.5%,5
GARCH–Merton,69.95,7.3415,-4.2017,-37.23,68.1%,3
Heston–Merton,55.01,5.6614,-2.4708,-22.17,61.8%,1


### Table MSFT-2  ·  Nov 2022  2022-11-18  ·  n = 34  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2022-11-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,62.41,3.3116,-1.2605,-15.13,69.9%,1
GARCH,62.95,3.3468,-1.2836,-15.37,71.2%,4
Heston,63.98,3.2743,-0.0458,+2.77,54.3%,6
Merton,62.54,3.3297,-1.2392,-14.78,67.4%,2
GARCH–Merton,62.88,3.3383,-1.2372,-14.70,67.3%,3
Heston–Merton,63.96,3.2667,-0.0490,+2.70,52.9%,5


### Table MSFT-3  ·  Dec 2022  2022-12-16  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2022-12-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,58.07,4.4568,-4.4568,-50.98,68.3%,6
GARCH,57.78,4.4426,-4.4426,-50.82,70.6%,4
Heston,34.81,2.5593,-2.5579,-29.41,54.7%,2
Merton,57.96,4.4486,-4.4486,-50.89,62.3%,5
GARCH–Merton,56.91,4.3801,-4.3801,-50.11,67.3%,3
Heston–Merton,34.73,2.5425,-2.5425,-29.21,53.5%,1


### Table MSFT-4  ·  Jan 2023  2023-01-20  ·  n = 34  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-01-20'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,27.87,2.1805,+0.3141,+3.68,90.5%,1
GARCH,27.97,2.1903,+0.3246,+3.80,89.5%,2
Heston,29.86,2.3193,+0.7234,+8.35,79.8%,5
Merton,28.06,2.1972,+0.3361,+3.94,87.0%,4
GARCH–Merton,27.98,2.1883,+0.3321,+3.90,87.3%,3
Heston–Merton,29.95,2.3275,+0.7365,+8.48,77.5%,6


### Table MSFT-5  ·  Feb 2023  2023-02-17  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-02-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,96.66,5.5490,-1.0468,-12.71,47.9%,3
GARCH,97.17,5.5906,-1.0680,-13.09,47.6%,6
Heston,95.08,5.3345,-0.0766,+3.61,43.1%,1
Merton,96.82,5.5592,-1.0332,-12.48,48.8%,4
GARCH–Merton,97.03,5.5751,-1.0357,-12.54,47.7%,5
Heston–Merton,95.15,5.3351,-0.0879,+3.42,43.0%,2


### Table MSFT-6  ·  Mar 2023  2023-03-17  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-03-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,89.98,5.0193,-5.0193,-88.16,19.7%,6
GARCH,89.46,4.9906,-4.9906,-87.61,19.3%,4
Heston,61.10,3.1716,-3.1716,-54.30,27.8%,2
Merton,89.95,5.0184,-5.0184,-88.15,19.0%,5
GARCH–Merton,89.08,4.9702,-4.9702,-87.22,19.4%,3
Heston–Merton,60.86,3.1530,-3.1530,-53.99,26.2%,1


### Table MSFT-7  ·  Apr 2023  2023-04-21  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-04-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,98.53,8.2487,-8.2487,-98.47,4.3%,3
GARCH,98.74,8.2598,-8.2598,-98.68,3.6%,6
Heston,87.25,7.5119,-7.5119,-86.11,12.6%,2
Merton,98.54,8.2490,-8.2490,-98.48,3.8%,4
GARCH–Merton,98.55,8.2494,-8.2494,-98.49,3.6%,5
Heston–Merton,87.15,7.5071,-7.5071,-85.99,12.6%,1


### Table MSFT-8  ·  May 2023  2023-05-19  ·  n = 34  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-05-19'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,8.1882,-8.1882,-100.00,0.0%,4
GARCH,100.00,8.1882,-8.1882,-100.00,0.0%,5
Heston,99.82,8.1748,-8.1748,-99.82,0.4%,2
Merton,100.00,8.1882,-8.1882,-100.00,0.0%,6
GARCH–Merton,100.00,8.1882,-8.1882,-100.00,0.0%,3
Heston–Merton,99.81,8.1738,-8.1738,-99.81,0.4%,1


### Table MSFT-9  ·  Jun 2023  2023-06-16  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-06-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,7.6081,-7.6081,-100.00,0.0%,3
GARCH,100.00,7.6081,-7.6081,-100.00,0.0%,4
Heston,99.82,7.5906,-7.5906,-99.82,0.3%,1
Merton,100.00,7.6081,-7.6081,-100.00,0.0%,5
GARCH–Merton,100.00,7.6081,-7.6081,-100.00,0.0%,6
Heston–Merton,99.83,7.5919,-7.5919,-99.83,0.3%,2


### Table MSFT-10  ·  Jul 2023  2023-07-21  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-07-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,10.3846,-10.3846,-100.00,0.0%,3
GARCH,100.00,10.3846,-10.3846,-100.00,0.0%,4
Heston,99.63,10.3469,-10.3469,-99.63,0.8%,1
Merton,100.00,10.3846,-10.3846,-100.00,0.0%,5
GARCH–Merton,100.00,10.3846,-10.3846,-100.00,0.0%,6
Heston–Merton,99.64,10.3476,-10.3476,-99.64,0.8%,2


### Table MSFT-11  ·  Aug 2023  2023-08-18  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-08-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,8.1132,-8.1132,-100.00,0.0%,3
GARCH,100.00,8.1132,-8.1132,-100.00,0.0%,4
Heston,99.97,8.1104,-8.1104,-99.97,0.0%,1
Merton,100.00,8.1132,-8.1132,-100.00,0.0%,5
GARCH–Merton,100.00,8.1132,-8.1132,-100.00,0.0%,6
Heston–Merton,99.98,8.1115,-8.1115,-99.98,0.0%,2


### Table MSFT-12  ·  Sep 2023  2023-09-15  ·  n = 34  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-09-15'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,7.2684,-7.2684,-100.00,0.0%,3
GARCH,100.00,7.2684,-7.2684,-100.00,0.0%,4
Heston,100.00,7.2682,-7.2682,-100.00,0.0%,1
Merton,100.00,7.2684,-7.2684,-100.00,0.0%,5
GARCH–Merton,100.00,7.2684,-7.2684,-100.00,0.0%,6
Heston–Merton,100.00,7.2683,-7.2683,-100.00,0.0%,2


## Summary S1


In [1]:
display(HTML('s1'))

Rank,Model,Mean RMSE%,Median RMSE%,# of 36 cells best
1,Heston–Merton,84.23,96.40,18
2,Heston,84.27,96.39,15
3,GARCH–Merton,90.94,99.84,0
4,GARCH,91.08,99.89,0
5,GBM,91.11,99.89,3
6,Merton,91.11,99.88,0


## Findings

- On mean percentage RMSE across the 36 company×window cells, Heston–Merton is the best overall model (mean RMSE% = 84.23; 18 of 36 cells).

- Second overall is Heston (mean RMSE% = 84.27; 15 cells).

- The ranking is not stable. Best-in-cell models across the 36 tables: GBM, Heston, Heston–Merton.

- SPY: lowest mean RMSE% is Heston–Merton (93.17); wins 6 of 12 windows.

- AAPL: lowest mean RMSE% is Heston–Merton (82.34); wins 7 of 12 windows.

- MSFT: lowest mean RMSE% is Heston–Merton (77.17); wins 5 of 12 windows.

- Jump models as a group have lower mean RMSE% (88.76) than the three no-jump models (88.82).

- These conclusions are conditional on the V3 filters, 7-day hourly calibration, nearest-ATM hourly sample, LSM with 10000 paths and 5-minute steps to Friday 15:59, and American-call quotes only. They are not a statement about European options or other tenors.
